### DISCUSSION TITLE
Can't use talk with HPB on Synology

### DISCUSSION THREAD

**Discussion by yoannhdev** [nextcloud/all-in-one#2269] [Questions] [answered]
### Steps to reproduce
1. Try to join conversation in talk -> connection failed
2. Remove HPB from admin config
3. Talk works fine

### Expected behavior 
Join talk's conversation with active HPB

### Actual behavior 
Talk's cant connect to a conversation when HPB is active 

### Host OS 
Synology DSM 7.1.1

#### Nextcloud AIO version 
v4.7.0

#### Current channel 
latest
#### Other valuable info 
```
curl -vvv https://$NC_DOMAIN:443/standalone-signaling/api/v1/welcome
```
```
*   Trying 90.30.208.205:443...
* Connected to cloud.yoannh.dev (90.30.208.205) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=cloud.yoannh.dev
*  start date: Apr  2 15:47:35 2023 GMT
*  expire date: Jul  1 15:47:34 2023 GMT
*  subjectAltName: host "cloud.yoannh.dev" matched cert's "cloud.yoannh.dev"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: GET]
* h2h3 [:path: /standalone-signaling/api/v1/welcome]
* h2h3 [:scheme: https]
* h2h3 [:authority: cloud.yoannh.dev]
* h2h3 [user-agent: curl/7.83.1]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x7f4659c361b0)
> GET /standalone-signaling/api/v1/welcome HTTP/2
> Host: cloud.yoannh.dev
> user-agent: curl/7.83.1
> accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 200 
< server: nginx
< date: Mon, 03 Apr 2023 22:24:40 GMT
< content-type: application/json; charset=utf-8
< content-length: 94
< x-spreed-signaling-features: audio-video-permissions, hello-v2, incall-all, mcu, simulcast, switchto, transient-data, update-sdp, welcome
< 
{"nextcloud-spreed-signaling":"Welcome","version":"de701dbcae52c77f3f98634cb9a61b15f79674ed"}
* Connection #0 to host cloud.yoannh.dev left intact
```
On docker logs, each time I try to join a conversation docker logs : hub.go:2314: Could not upgrade request from 90.30.208.205: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header

**Comment by szaimen** [comment, marked as answer]
> On docker logs, each time I try to join a conversation docker logs : hub.go:2314: Could not upgrade request from 90.30.208.205: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header

You have found your problem. Apparently are websocket connections not enabled for your domain. So probably something that you need to fix in your Reverse Proxy.

**Reply by yoannhdev** [reply]
Thanks for your answer. Actually there is an option "add a websocket" on Synology reverse proxy... everything works fine now.

**Reply by szaimen** [reply]
Great that you found the solution!

Could you possibly post a screenshot of this setting that we could put into our reverse proxy documentation? Thanks a lot! :)
